packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2414,17 +2414,17 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     final bool childrenMergeIntoParent = mergeIntoParent || config.isMergingSemanticsOfDescendants;
 
     visitChildrenForSemantics((RenderObject renderChild) {
-      final _SemanticsFragment fragment = renderChild._getSemanticsForParent(
+      final _SemanticsFragment parentFragment = renderChild._getSemanticsForParent(
```

Comment:
Curious: what variable was this shadowing?

Reply:
This variable wasn't the one shadowing. It was shadowed by the identically named variable defined in the `for` loop below. So if, for example, you renamed the loop variable you'd have a bug and no feedback from the analyzer/compiler whatsoever.